go-backend/db/organization.go

```diff
@@ -0,0 +1,216 @@
+	ContactEmail             string    `db:"contact_email" json:"email"`
+	DomainName               string    `db:"domain_name" json:"domain_name"`
+	SubscriptionStatus       int       `db:"subscription_status" json:"subscription_status"`
+	SubscriptionValidUpto    time.Time `db:"subscription_valid_upto" json:"subscription_valid_upto"`
+	Hi5Limit                 int    `db:"hi5_limit" json:"hi5_limit"`
+	Hi5QuotaRenewalFrequency string    `db:"hi5_quota_renewal_frequency" json:"hi5_quota_renewal_frequency"`
+	Timezone                 string    `db:"timezone" json:"timezone"`
+	CreatedBy                int    `db:"created_by" json:"_"`
+	CreatedOn                time.Time `db:"created_on" json:"_"`
+	UpdatedBy                int    `db:"updated_by" json:"_"`
+	UpdatedOn                time.Time `db:"updated_on" json:"_"`
+}
+
+//TODO how to declare this as reusable
+type ErrorResponse struct {
+	Code 	string `json:"code"`
+	Message	string `json:"message"`
+	Fields map[string]string `json:"fields"`
+}
+
+func (org *Organization) ValidateOrganization() (errorResponse map[string]ErrorResponse, valid bool) {
+
+	valid = true
+	fieldErrors := make(map[string]string)
+	var validEmail, _ = regexp.Compile(emailRegex)
+	var validDomain, _ = regexp.Compile(domainRegex)
+	
+	if org.Name == "" {
+		valid = false
+		fieldErrors["name"] = "Can't be blank"
+	}
+
+	if !validEmail.MatchString(org.ContactEmail) {
+		valid = false
+		fieldErrors["email"] = "Please enter a valid email"
+	}
+
+	if !validDomain.MatchString(org.DomainName) {
+		fieldErrors["domain_name"] = "Please enter valid domain"
+		valid = false
+	}
+
+	errorResponse = map[string]ErrorResponse{"error": ErrorResponse{
+		Code: "invalid_data",
+		Message: "Please provide valid organization data",
+		Fields: fieldErrors,
+	},
+}
+
+	//TODO: Ask what other validations are expected
+	
+	return
+}
+
+func (s *pgStore) ListOrganizations(ctx context.Context) (organizations []Organization, err error) {
+
+	err = s.db.Select(&organizations, ListOrganizationsQuery)
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error listing organizations")
+		return
+	}
+
+	return
+}
+
+func (s *pgStore) CreateOrganization(ctx context.Context, org Organization) (err error) {
+
+	_, err = s.db.Exec(
+		CreateOrganizationQuery,
+		org.Name,
+		org.ContactEmail,
+		org.DomainName,
+		org.SubscriptionStatus,
+		org.SubscriptionValidUpto,
+		org.Hi5Limit,
+		org.Hi5QuotaRenewalFrequency,
+		org.Timezone,
+		org.CreatedBy,
+		org.UpdatedBy,
+	)
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error creating organization")
+		return
+	}
+
+	return
+}
+
+func (s *pgStore) UpdateOrganization(ctx context.Context, reqOrganization Organization, organizationID int) (updatedOrganization Organization, err error) {
+
+		var dbOrganization Organization
+		err = s.db.Get(&dbOrganization, GetOrganizationQuery, organizationID)
+
+		reqOrganization.CreatedOn = dbOrganization.CreatedOn
```

Comment:
your update query is not allowing to update the CreatedOn, CreatedBy fields. The query does not have these fields in list of fields allowed for update.
What is the benefit of this assignment then ?

Reply:
I see maybe you guys are returning a value that's why

Reply:
We did that because while updating we don't want to modify created_on and created_by fields and if we don't assign explicitly it takes default struct zero values

Reply:
It does not matters for you case because your update query explicitly specifies what values are being updated. So even if the `reqOrganization` has zero values for `CreatedOn`, `CreatedBy`, we dont care as we are not updating those values